include/mata/nfa/algorithms.hh

```diff
@@ -112,6 +132,7 @@ Simlib::Util::BinaryRelation compute_relation(
  * @param[out] prod_map Mapping of pairs of the original states (lhs_state, rhs_state) to new product states.
  * @return NFA as a product of NFAs @p lhs and @p rhs with ε-transitions preserved.
  */
+// all parameters needed/used?
```

Comment:
We can ask at the mejtyng what the uses can be, and remove it if there is nothing concrete.

Reply:
Maybe really just putting the one struct of the special type as the optional parameter there, with some arbitrary initial choice of optional parameters as members? That would allow us to change the insides of the struct later without having to modify the interface?

Reply:
Note that here we are talking about output parameters. When we pass the parameters, we normally do not want to be able to change the parameter class inside the function (we want to pass it by const reference). Hence, mixing the input, output, and input/output parameters in a single class would be confusing and error-prone, I believe.

Reply:
as in the above comment, for these maps, I would probably put them to the key-value store

Reply:
See https://github.com/VeriFIT/mata/pull/277#discussion_r1281404329 for what problems I find with this approach.